src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,86 @@
+import cl from 'classnames';
+import React, { FormEvent, useState } from 'react';
+import { addComment } from '../api/api';
+import { pause } from '../utils/methods';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  postId: number;
+  onAddComment: (v: Comment) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onAddComment }) => {
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setBodyError] = useState(false);
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleAddChange = async () => {
+    setNameError(!name);
+    setEmailError(!email);
+    setBodyError(!body);
+
+    if (!name || !email || !body) {
+      return;
+    }
+
+    const newComment = {
+      postId,
+      name,
+      email,
+      body,
+    };
+
+    setIsLoading(true);
+    try {
+      await pause();
+
+      const res = await addComment(newComment);
+
+      onAddComment(res);
+      setBody('');
+    } catch {
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleAuthorChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setNameError(false);
+  };
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setEmailError(false);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setBody(e.target.value);
+    setBodyError(false);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+  };
```

Comment:
The handleSubmit function currently prevents default form submission but does not trigger the comment addition logic. Ensure that the form submission correctly calls handleAddChange or another function to add the comment.